File: Java/Tests/jdi-uitest-webtests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/w3cSite/FramePage.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.w3cSite;

import com.epam.jdi.uitests.web.selenium.elements.common.Button;
import com.epam.jdi.uitests.web.selenium.elements.composite.WebPage;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.Frame;
import org.openqa.selenium.support.FindBy;

import java.util.Iterator;

/**
 * Created by Maksim_Palchevskii on 8/17/2015.
 */

public class FramePage extends WebPage {
    @Frame(id = "iframeResult")
    public FrameSection frame;

    @FindBy(partialLinkText = "Try it Yourself")
    public Button tryIt = new Button() {
        @Override
        public void clickAction() {
            super.clickAction();
            Iterator<String> i = getDriver().getWindowHandles().iterator();
            i.next();
            getDriver().switchTo().window(i.next());
        }
    };
}

File: Python/Test/jdi_uitests_webtests/main/page_objects/w3c_site/frame_page.py
from JDI.web.selenium.elements.api_interact.find_element_by import By
from JDI.web.selenium.elements.common.button import Button
from JDI.web.selenium.elements.composite.web_page import WebPage
from Test.jdi_uitests_webtests.main.page_objects.w3c_site.frame_section import FrameSection


class FramePage(WebPage):
    def __init__(self, url, title=None, domain=None):
        super(FramePage, self).__init__(url, title, domain)

    i_frame = FrameSection(By.id("iframeResult"))

    try_it_button = Button(By.link_text("Try it Yourself »"))